### PR TITLE
Import user thumbnail

### PR DESCRIPTION
**Pull request by krokrob**
This solves https://github.com/lewagon/www/issues/268 
`user.thumbnail(:thumb)` is systematically called when a thumbnail is needed.
As `user.thumbnail(:thumb)` refers to `user.picture.url(:thumb)`, we have to seed `user.picture` if it does not exists. I proposed a callback after_create for `User` for new `users` and this script for existing ones: 

```
User.find_each do |user|
  unless user.picture.exists?
    user.picture = user.gravatar_url
    user.save
  end
end
```

### CODE REVIEW

**Comment by ssaunier** [comment]
It's a good start, but for me it won't be enough. Basically we should move the pictures to cloudinary so that we can do on the fly size change. The `www` website uses `` which are small than the picture real size, that's why google complains (too much pixels!) If the APi could provide a base URL to which we can append some cloudinary transformation (width / crop / etc), that would be awesome. We could start by moving the user picture to cloudinary (using attachinary).

**Comment by krokrob** [comment]
@ssaunier: I am sure this is not DRY and we have to make decisions about naming some redondant transformation BUT every call to an user image is now optimized 😉  

After merge and deploy and migrate, we must run:

```
User.find_each do |user|
  if user.picture.exists?
    user.photo = open(user.picture.url)
  else
    user.photo = open(user.gravatar_url)
  end
  user.save!
end
```

File: config/cloudinary.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+cloud_name: wagon
+api_key: '141994738762271'
+api_secret: obhLkFVa13NTl3FRqiq2WhktydA
```
Comment (ssaunier) [inline config/cloudinary.yml]: arf pas cool on a du secret qui a fuité dans GitHub. Je vais refénérer une clé. Pourquoi tu as besoin de ce fichier ? Pas suffisant le `config/application.yml`?

Comment (krokrob) [inline config/cloudinary.yml]: Oh sh..., désolé pour cette erreur de débutant. 👎

**Comment by krokrob** [comment]
@ssaunier alumni API now sends `photo_path` associated with `user`. Don't merge before I adapt www 😉

File: app/models/user.rb [hunk 2]
```diff
@@ -141,10 +146,14 @@ def name
     )
   end
 
-  def thumbnail(style = :medium)
-    picture.exists? ? picture.url(style) : gravatar_url
-  rescue SocketError
-    gravatar_url
+  def thumbnail(options={})
+    self.photo.nil? ? gravatar_url : cloudinary_url(self.photo.path, options)
```
Comment (ssaunier) [inline app/models/user.rb]: Pas besoin de tous ces `self` en fait.

**Comment by ssaunier** [comment]
I feel like we'll need to run this in console then:

```
require "open-uri"
User.all.each do |user|
  if user.picture.exists?
    user.photo = open(user.picture.url)
  else
    user.photo = open(user.gravatar_url)
  end
  user.save
end
```

That way we'll upload all the user pictures to Cloudinary. Then we will be able to get rid of paperclip `:picture` on `User`.

Right?

File: app/models/user.rb [hunk 3]
```diff
@@ -169,10 +178,17 @@ def github_nickname_exists?
...
       self.email = github_user.email if email.blank?
     rescue Octokit::NotFound => e
       errors.add :github_nickname, "This github user does not exist"
     end
   end
+
+  def set_default_picture
+    unless self.gravatar_url.nil?
+      self.photo_url = self.gravatar_url
+      self.save
```
Comment (ssaunier) [inline app/models/user.rb]: Should be in a `before_create`, and get rid of `self.save`

File: app/models/user.rb [hunk 4]
```diff
@@ -169,10 +178,17 @@ def github_nickname_exists?
...
+      self.gravatar_url = github_user.avatar_url if gravatar_url.blank?
       self.email = github_user.email if email.blank?
     rescue Octokit::NotFound => e
       errors.add :github_nickname, "This github user does not exist"
     end
   end
+
+  def set_default_picture
+    unless self.gravatar_url.nil?
+      self.photo_url = self.gravatar_url
```
Comment (ssaunier) [inline app/models/user.rb]: No need for `self.gravatar_url`. But need for `self.photo_url` (talking about `self`)

**Pull request merged**